src/ModuleResolver.ts

```diff
@@ -322,8 +322,24 @@ export class ModuleResolver implements ModuleResolverInterface {
     return resolvedIgnorePath;
   }
 
+  private adjustFileNameForPrettierVersion3_1_1(
+    prettierInstance: { version: string | null },
+    fileName: string
+  ) {
+    if (!prettierInstance.version) {
+      return fileName;
+    }
+    // Avoid https://github.com/prettier/prettier/pull/15363
+    const isGte3_1_1 = semver.gte(prettierInstance.version, "3.1.1");
+    if (isGte3_1_1) {
+      return path.join(fileName, "tmp");
```

Comment:
@fisker  `fileName` is opened project directory path. And it has temporary file name. Do you have a better plan?

Reply:
Perhaps a more random name, such as UUID, should be used?

Reply:
Can't we use the real file path?

Reply:
Prettier use `noop.js` for import from directory https://github.com/prettier/prettier/blob/41baabd62734a4b8f4065294cad873af3b46dbb3/src/utils/import-from-directory.js#L6

Reply:
> Can't we use the real file path?

Maybe it can be done, but I'm not familiar with prettier-vscode and don't know what the expected behavior is...
So I thought it would be quicker if we could take an approach like this PR.

Reply:
I've updated to use `noop.js` instead of `tmp`.

Reply:
@fisker ping

Reply:
I'm  not familiar with, you can merge if you think it will work, but better wait for @ntotten ?